test/populate_test/bdl_commands_test/window_test.py

```diff
@@ -21,15 +21,109 @@ def setUp(self):
         self.exterior_wall = ExteriorWall("Exterior Wall 1", self.space, self.rmd)
         self.window = Window("Window 1", self.exterior_wall, self.rmd)
 
-    # @patch("rpd_generator.bdl_structure.base_node.BaseNode.get_output_data")
-    # def test_populate_data_with_window(self, mock_get_output_data):
-    #     mock_get_output_data.return_value = {}
-    #
-    #     self.window.keyword_value_pairs = {}
-    #
-    #     self.window.populate_data_elements()
-    #     self.window.populate_data_group()
-    #
-    #     expected_data_structure = {}
-    #
-    #     self.assertEqual(expected_data_structure, self.window.window_data_structure)
+    def test_populate_data_with_window(self):
+        self.window.keyword_value_pairs = {
+            BDL_WindowKeywords.HEIGHT: 4,
+            BDL_WindowKeywords.WIDTH: 3,
+            BDL_WindowKeywords.FRAME_WIDTH: 2,
+            BDL_WindowKeywords.LEFT_FIN_D: 1.0,
+            BDL_WindowKeywords.OVERHANG_D: 1.5,
+            BDL_WindowKeywords.SHADING_SCHEDULE: "Schedule",
+            BDL_WindowKeywords.WIN_SHADE_TYPE: "MOVABLE-SHADE"
```

Comment:
the WIN-SHADE-TYPE keyword has limited options available to the user. Here is the help-text from eQUEST.

> WIN-SHADE-TYPE
> Takes a code-word giving the type of shading device when a shading device is present on the window for sun and/or glare control. If SHADING-SCHEDULE is not assigned to a window, the window will be considered to have no shading device and WIN-SHADE-TYPE will be ignored. 
> 
> MOVABLE-INTERIOR      (the default) interior shade that can be retracted, such as drapes or Venetian blinds.
> 
> MOVABLE-EXTERIOR      exterior shade that can be retracted.
> 
> FIXED-INTERIOR           interior shade that cannot be retracted.
> 
> FIXED-EXTERIOR          exterior shade that cannot be retracted.
>

Reply:
Best practice for using these enumerations is to add them to the `rpd_generator/bdl_structure/bdl_enumerations/bdl_enums.py` file to 1) avoid using magic strings in the code and 2) help to build out a complete set of the enumerations that are used in eQUEST for future use.

Reply:
After reviewing window.py again I was not happy with my old work so I did some cleaning up which included adding these shade type enumerations so they can be used in the test file too.